src/set_theory/ordinal/arithmetic.lean

```diff
@@ -1217,6 +1238,13 @@ bsup_eq_zero_iff.2 (λ i hi, by { subst ho, exact (ordinal.not_lt_zero i hi).eli
 theorem bsup_const {o : ordinal} (ho : o ≠ 0) (a : ordinal) : bsup o (λ _ _, a) = a :=
 le_antisymm (bsup_le (λ _ _, le_rfl)) (le_bsup _ 0 (ordinal.pos_iff_ne_zero.2 ho))
 
+theorem bsup_unique {o : ordinal} (ho : o = 1) (f : Π a < o, ordinal) :
+  bsup o f = f 0 (by convert zero_lt_one) :=
```

Comment:
I think this would be better without `0`
```suggestion
theorem bsup_unique (f : Π a < 1, ordinal) :
  bsup 1 f = f 0 (by convert zero_lt_one) :=
```

Reply:
I tried this for a bit with `bsup_empty`, but I found that doing this makes it very annoying to use these theorems. `heq` hell, if you will. 

The other direction is super simple, though. The theorem as you write it down is just `bsup_unique rfl`.

Reply:
```lean
example {o : ordinal} (f : Π a < succ 0, ordinal) : bsup _ f = f 0 sorry := sorry
```
Even though `succ 0 = 1`, it becomes a hassle to rewrite it.

Reply:
I could still add your lemma alongside, though.

Reply:
It becomes a lot easier if you add a `congr` lemma:
```lean
@[simp]
theorem bsup_one (f : Π a < (1 : ordinal), ordinal) :
  bsup 1 f = f 0 (zero_lt_one) :=
by simp_rw [←sup_eq_bsup, sup_unique, family_of_bfamily, family_of_bfamily', typein_lt_one_out]

@[congr]
lemma bsup_congr {o₁ o₂ : ordinal} (f : Π a < o₁, ordinal) (ho : o₁ = o₂) :
  bsup o₁ f = bsup o₂ (λ a h, f a (h.trans_eq ho.symm)) :=
by subst ho

example {o : ordinal} (f : Π a < succ 0, ordinal) : bsup _ f = f 0 (succ_pos _) :=
by simp
```

Reply:
What's the status here?